src/app/app.component.ts

```diff
@@ -1,51 +1,85 @@
 import { Component } from '@angular/core';
 import { Task } from './model/task.model';
 import { isEmpty } from 'underscore';
+import { WeatherService } from './services/weather.service';
+import { OnInit } from '@angular/core/src/metadata/lifecycle_hooks';
 
 @Component({
   selector: 'app-root',
   templateUrl: 'app.component.html',
   styleUrls: ['app.component.css'],
 })
-export class AppComponent {
+export class AppComponent implements OnInit{
 
+  weather: Object;
   activatedTasks: Task[] = [];
   completedTasks: Task[] = [];
   currentTaskInTextInput: String;
+  newCityId: String;
+  dataLoaded: boolean = false;
+  newCityIdEntered: boolean;
 
-  addNewtask(newTaskName: String) {
+  constructor(private weatherService: WeatherService) { }
+
+  ngOnInit(): void {
+    this.getCityWeatherAsync();
+  }
+
+  ngAfterViewChecked(): void {
+    if (this.newCityIdEntered) {
+      this.getCityWeatherAsync();
+    }
+  }
+
+  getCityWeatherAsync(): void {
```

Comment:
Clean Code : don't use Async word in the function name, you can return a function instead
getCityWeather(): Promise<any>